lib/workspace.js

```diff
@@ -203,10 +209,11 @@ module.exports = {
         if (state == 'uninitialized' || state == 'stopped') {
             if (state == 'uninitialized') {
                 await module.exports.initialize(workspace_id);
-                await module.exports.updateState(workspace_id, 'stopped');
+                await module.exports.updateState(workspace_id, 'stopped', 'Initialization complete');
             }
-            await module.exports.updateState(workspace_id, 'launching');
+            await module.exports.updateState(workspace_id, 'launching', 'Assigning workspace host');
             await module.exports.assignHost(workspace_id);
+            await module.exports.updateMessage(workspace_id, 'Sending launch command to host');
```

Comment:
Or should we set `launching` state again with a new `message`?

Reply:
From a state-machine perspective, we aren't changing state. I think it's best to conceptually separate state changes (which mean something to the code execution) and message updates (purely for display).